link-grammar/parse/prune.c

```diff
@@ -1129,17 +1129,169 @@ static int pp_prune(Sentence sent, Parse_Options opts)
 	return total_deleted;
 }
 
+/**
+ * Discard disjuncts that cannot connect due to cross links.
+ * This is performed only for null_count==0, because any word can then
+ * may not have links.
+ * Optional words are totally ignored because they don't have to link.
+ * FIXME: Only the first connector on each side is checked.
+ *
+ * Algo:
+ * 1. For each word (w) find the nearest words (according to the shallow
+ * connectors of its disjuncts) to its left (nl) and right (nr) to * which
+ * a link must exist. I.e., a link from w must exist to nr or a word after
```

Comment:
what's an example of a link that "must" exist?

Reply:
See my numerical example below. An left-going ink of a length of at least 5 must exist there.